app/src/main/java/com/nononsenseapps/feeder/ui/compose/feed/FeedScreen.kt

```diff
@@ -495,145 +509,393 @@ fun FeedScreen(
+                                                role = Role.Checkbox
+                                            },
+                                )
+                                DropdownMenuItem(
+                                    onClick = {
+                                        filterCallback.setRead(!viewState.filter.read)
+                                        // Closing it is important for accessibility
+                                        onShowFilterMenu(false)
+                                    },
+                                    leadingIcon = {
+                                        Icon(
+                                            when (viewState.filter.read) {
+                                                true -> Icons.Default.CheckBox
+                                                false -> Icons.Default.CheckBoxOutlineBlank
+                                            },
+                                            contentDescription = null,
+                                        )
+                                    },
+                                    text = {
+                                        Text(stringResource(id = R.string.read_adjective))
+                                    },
+                                    modifier =
+                                        Modifier
+                                            .safeSemantics {
+                                                stateDescription =
+                                                    when (viewState.filter.read) {
+                                                        true -> context.getString(androidx.compose.ui.R.string.selected)
+                                                        else -> context.getString(androidx.compose.ui.R.string.not_selected)
+                                                    }
+                                                role = Role.Checkbox
+                                            },
+                                )
+                                // Hidden button for TalkBack
+                                DropdownMenuItem(
+                                    onClick = {
+                                        onShowFilterMenu(false)
+                                    },
+                                    text = {},
+                                    modifier =
+                                        Modifier
+                                            .height(0.dp)
+                                            .safeSemantics {
+                                                contentDescription = closeMenuText
+                                                role = Role.Button
+                                            },
+                                )
+                            }
+                        }
+                    }
+                }
+
+                PlainTooltipBox(tooltip = { Text(stringResource(R.string.search_noun)) }) {
```

Comment:
IMO the search button should definitely be the first button in the toolbar so move that up